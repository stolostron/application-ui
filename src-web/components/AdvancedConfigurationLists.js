/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import ResourceList from './common/ResourceList'
import ResourceTableModule from './common/ResourceTableModuleFromProps'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import getResourceDefinitions from '../definitions'
import { makeGetVisibleTableItemsSelector } from '../reducers/common'
import QuerySwitcher from './common/QuerySwitcher'

const AdvancedConfigurationLists = (props) => {

  const resourceType = RESOURCE_TYPES.QUERY_APPLICATIONS
  const staticResourceData = getResourceDefinitions(resourceType)
  const getVisibleResources = makeGetVisibleTableItemsSelector(resourceType)

  return (
    <ResourceList
      {...props}
      resourceType={resourceType}
      staticResourceData={staticResourceData}
      getVisibleResources={getVisibleResources}
      modules={[<ResourceTableModule key="deployments" definitionsKey="deploymentKeys" />]}
    >
      <QuerySwitcher
        key='switcher'
        queryParam="resource"
        options={
              [{ id: 'subscription', contents: 'Subscriptions' },
                { id: 'placementrule', contents: 'Placement rules'},
                {id: 'channel', contents: 'Channels'}]}
        defaultOption="subscription"
      />
    </ResourceList>
  )}

export default AdvancedConfigurationLists
