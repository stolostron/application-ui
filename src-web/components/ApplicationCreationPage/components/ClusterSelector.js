/*******************************************************************************
 * Licensed Materials - Property of IBM
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import resources from '../../../../lib/shared/resources'
import {
  Accordion,
  AccordionItem,
  Checkbox,
  Icon,
  TextInput
} from 'carbon-components-react'
import { getSourcePath, removeVs } from '../../TemplateEditor/utils/utils'
import Tooltip from '../../TemplateEditor/components/Tooltip'
import msgs from '../../../../nls/platform.properties'
import _ from 'lodash'

resources(() => {
  require('./style.scss')
})

export class ClusterSelector extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    controlId: PropTypes.string,
    handleChange: PropTypes.func,
    locale: PropTypes.string
  };

  constructor(props) {
    super(props)
    this.state = {}
    if (_.isEmpty(this.props.control.active)) {
      if (
        !this.props.control.showData ||
        this.props.control.showData.length === 0
      ) {
        this.props.control.active = {
          mode: true,
          clusterLabelsList: [
            { id: 0, labelName: '', labelValue: '', validValue: false }
          ],
          clusterLabelsListID: 1
        }
      } else {
        //display existing placement rule
        this.props.control.active = {
          mode: true,
          clusterLabelsList: this.props.control.showData,
          clusterLabelsListID: this.props.control.showData.length
        }
      }
    }
    this.props.control.validation = this.validation.bind(this)
  }

  render() {
    if (!this.props.control.active) {
      this.props.control.active = {
        mode: true,
        clusterLabelsList: [
          { id: 0, labelName: '', labelValue: '', validValue: false }
        ],
        clusterLabelsListID: 1
      }
    }
    const { controlId, locale, control } = this.props
    const { name, active, validation = {} } = control
    const modeSelected = active && active.mode === true
    const isReadOnly = _.get(this.props, 'control.showData', []).length > 0

    return (
      <React.Fragment>
        <div className="creation-view-controls-labels">
          <div>
            {name}
            {validation.required ? (
              <div className="creation-view-controls-required">*</div>
            ) : null}
            <Tooltip control={control} locale={locale} />
          </div>

          <div className="clusterSelector-container">
            <Checkbox
              className="clusterSelector-checkbox"
              name="clusterSelector-checkbox"
              checked={modeSelected}
              disabled={isReadOnly}
              id={`clusterSelector-checkbox-${controlId}`}
              labelText={msgs.get(
                'tooltip.creation.app.settings.clusterSelector',
                locale
              )}
              onClick={this.handleChange.bind(this)}
            />

            <Accordion>
              <AccordionItem open={modeSelected} title={''}>
                <div className="clusterSelector-labels-section">
                  <div className="labels-descr">
                    {msgs.get(
                      'creation.app.settings.selectorClusters.config',
                      locale
                    )}
                  </div>

                  <div
                    className="labels-section"
                    id={`clusterSelector-labels-section-${controlId}`}
                  >
                    {this.renderClusterLabels(control, isReadOnly, controlId)}
                    <div
                      className={`add-label-btn ${
                        isReadOnly ? 'btn-disabled' : ''
                      }`}
                      tabIndex="0"
                      role={'button'}
                      onClick={() => this.addLabelToList(control, !isReadOnly)}
                      onKeyPress={this.addLabelKeyPress.bind(this)}
                    >
                      <Icon
                        name="icon--add--glyph"
                        fill="#3d70b2"
                        description=""
                        className="add-label-btn-icon"
                      />
                      <div className="add-label-btn-text">
                        {msgs.get(
                          'creation.app.settings.selectorClusters.prop.add',
                          locale
                        )}
                      </div>
                    </div>
                  </div>
                </div>
              </AccordionItem>
            </Accordion>
          </div>
        </div>
      </React.Fragment>
    )
  }

  validation(exceptions) {
    const { control, locale, controlId } = this.props
    if (_.get(control, 'active.mode', false)) {
      if (Object.keys(control.active.clusterLabelsList).length === 0) {
        //no cluster labels set
        exceptions.push({
          row: 1,
          text: msgs.get('creation.missing.clusterSelector.value', locale),
          type: 'error',
          controlId: `clusterSelector-labels-section-${controlId}`
        })
      }

      control.active.clusterLabelsList.map(item => {
        const { id, labelName, labelValue } = item
        const invalidLabel = !labelName || labelName.length === 0
        const invalidValue = !labelValue || labelValue.length === 0

        // Add exception if no input for labels or values
        if (invalidLabel) {
          exceptions.push({
            row: 1,
            text: msgs.get('creation.missing.clusterSelector.label', locale),
            type: 'error',
            controlId: `labelName-${id}`
          })
        }
        if (invalidValue) {
          exceptions.push({
            row: 1,
            text: msgs.get('creation.missing.clusterSelector.value', locale),
            type: 'error',
            controlId: `labelName-${id}`
          })
        }
      })
    }
  }

  renderClusterLabels = (control, isReadOnly, controlId) => {
    if (!_.get(control, 'active.clusterLabelsList')) {
      return ''
    }
    return (
      control.active &&
      control.active.clusterLabelsList.map(item => {
        const { id, labelName, labelValue, validValue } = item
        const invalidLabel = !labelName || labelName.length === 0
        const exceptionLabel = msgs.get(
          'creation.missing.clusterSelector.label.field'
        )
        const invalidValue = !labelValue || labelValue.length === 0
        const exceptionValue = msgs.get(
          'creation.missing.clusterSelector.value.field'
        )

        if (validValue || id === 0) {
          return (
            <React.Fragment key={id}>
              <div className="matching-labels-container">
                <div className="matching-labels-input">
                  <TextInput
                    id={`labelName-${id}-${controlId}`}
                    invalid={invalidLabel}
                    invalidText={exceptionLabel}
                    name="labelName"
                    className="text-input"
                    labelText={
                      id === 0
                        ? `${msgs.get('clusterSelector.label.field.ui')}*`
                        : ''
                    }
                    value={labelName === '' ? '' : labelName}
                    placeholder={msgs.get(
                      'clusterSelector.label.placeholder.field'
                    )}
                    disabled={isReadOnly}
                    onChange={this.handleChange.bind(this)}
                  />
                </div>
                <div className="matching-labels-input">
                  <TextInput
                    invalid={invalidValue}
                    invalidText={exceptionValue}
                    id={`labelValue-${id}-${controlId}`}
                    name="labelValue"
                    className="text-input"
                    labelText={
                      id === 0
                        ? `${msgs.get('clusterSelector.value.field.ui')}*`
                        : ''
                    }
                    value={labelValue === '' ? '' : labelValue}
                    placeholder={msgs.get(
                      'clusterSelector.value.placeholder.field'
                    )}
                    disabled={isReadOnly}
                    onChange={this.handleChange.bind(this)}
                  />
                </div>

                {id !== 0 ? ( // Option to remove added labels
                  <div
                    id={id}
                    className="remove-label-btn"
                    tabIndex="0"
                    role={'button'}
                    onClick={() => this.removeLabelFromList(control, item)}
                    onKeyPress={this.removeLabelKeyPress.bind(this)}
                  >
                    <Icon
                      name="icon--close--glyph"
                      fill="#3d70b2"
                      className="remove-label-btn-icon"
                    />
                  </div>
                ) : (
                  ''
                )}
              </div>
            </React.Fragment>
          )
        }
        return ''
      })
    )
  };

  addLabelToList = (control, modeSelected) => {
    if (modeSelected) {
      // Create new "label" item
      control.active.clusterLabelsList.push({
        id: control.active.clusterLabelsListID,
        labelName: '',
        labelValue: '',
        validValue: true
      })
      control.active.clusterLabelsListID++

      // Update UI
      this.forceUpdate()
    }
  };

  addLabelKeyPress = e => {
    if (e.type === 'click' || e.key === 'Enter') {
      this.addLabelToList(this.props.control)
    }
  };

  removeLabelFromList = (control, item) => {
    // Removed labels are no longer valid
    control.active.clusterLabelsList[item.id].validValue = false

    // Update UI and yaml editor
    this.forceUpdate()
    this.handleChange({})
  };

  removeLabelKeyPress = e => {
    if (e.type === 'click' || e.key === 'Enter') {
      this.removeLabelFromList(this.props.control, { id: e.target.id })
    }
  };

  handleChange(event) {
    const { control, handleChange } = this.props
    let targetName = ''
    try {
      targetName = event.target.name
    } catch (e) {
      targetName = ''
    }

    if (targetName) {
      const { active } = control
      if (targetName === 'clusterSelector-checkbox') {
        active.mode = event.target.checked ? true : false
      } else {
        const { clusterLabelsList } = active
        const labelID = parseInt(event.target.id.split('-')[1], 0)
        if (targetName === 'labelName') {
          clusterLabelsList[labelID].labelName = event.target.value
        } else if (targetName === 'labelValue') {
          clusterLabelsList[labelID].labelValue = event.target.value
        }
        clusterLabelsList[labelID].validValue = true
      }
    }

    handleChange(control)
  }
}

export default ClusterSelector

export const summarize = (control, controlData, summary) => {
  const { clusterLabelsList } = control.active || {}
  if (
    clusterLabelsList &&
    _.get(control, 'type', '') !== 'hidden' &&
    _.get(control, 'active.mode')
  ) {
    clusterLabelsList.forEach(({ labelName, labelValue }) => {
      if (labelName && labelValue) {
        summary.push(`${labelName}=${labelValue}`)
      }
    })
  }
}

export const reverse = (control, templateObject) => {
  if (!control.active) {
    let matchLabels
    const local = _.get(
      templateObject,
      getSourcePath(
        'PlacementRule[0].spec.clusterSelector.matchLabels.local-cluster'
      )
    )
    if (!local) {
      matchLabels = _.get(
        templateObject,
        getSourcePath('PlacementRule[0].spec.clusterSelector.matchLabels')
      )
      if (!matchLabels) {
        matchLabels = _.get(
          templateObject,
          getSourcePath('PlacementRule[0].spec.clusterLabels.matchLabels')
        )
      }
    }
    if (matchLabels) {
      matchLabels = removeVs(matchLabels)
      if (matchLabels) {
        const clusterLabelsList = Object.entries(matchLabels).map(
          ([labelName, labelValue], id) => {
            return {
              id,
              labelName,
              labelValue,
              validValue: true
            }
          }
        )
        control.active = {
          mode: true,
          clusterLabelsList,
          clusterLabelsListID: clusterLabelsList.length
        }
      }
    } else {
      const clusterLabelsList = [
        { id: 0, labelName: '', labelValue: '', validValue: false }
      ]
      control.active = {
        mode: false,
        clusterLabelsList,
        clusterLabelsListID: 1
      }
    }
  }
}
