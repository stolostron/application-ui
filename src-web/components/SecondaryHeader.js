/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Breadcrumb, Tabs, Tab, Button } from 'carbon-components-react'
import { DetailPageHeader } from 'carbon-addons-cloud-react'
import resources from '../../lib/shared/resources'
import { withRouter, Link } from 'react-router-dom'
import msgs from '../../nls/platform.properties'
import SecondaryHeaderTooltip from './SecondaryHeaderTooltip'
import classNames from 'classnames'

resources(() => {
  require('../../scss/secondary-header.scss')
})

export class SecondaryHeader extends React.Component {
  constructor(props) {
    super(props)
    this.renderBreadCrumb = this.renderBreadCrumb.bind(this)
    this.renderTabs = this.renderTabs.bind(this)
    this.renderTooltip = this.renderTooltip.bind(this)
    this.renderLinks = this.renderLinks.bind(this)

    this.state = {
      shadowPresent: false
    }
  }

  componentDidMount() {
    window.addEventListener('scroll', this.listenToScroll)
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.listenToScroll)
  }

  listenToScroll = () => {
    if (window.scrollY > 0.1 && this.state.shadowPresent === false) {
      this.setState({ shadowPresent: true })
    } else if (window.scrollY <= 0.1 && this.state.shadowPresent === true) {
      this.setState({ shadowPresent: false })
    }
  };

  render() {
    const { tabs, title, breadcrumbItems, links, actions } = this.props
    const { locale } = this.context
    if (
      (tabs && tabs.length > 0) ||
      (breadcrumbItems && breadcrumbItems.length > 0)
    ) {
      return (
        <div
          className={classNames({
            'secondary-header-wrapper': true,
            'with-tabs': tabs && tabs.length > 0,
            'with-breadcrumbs': breadcrumbItems && breadcrumbItems.length > 0
          })}
          role="region"
          aria-label={title}
        >
          <div
            className={`secondary-header ${
              actions && !tabs ? 'detailed-header-override' : ''
            }`}
          >
            <React.Fragment>
              <DetailPageHeader
                hasTabs={true}
                title={decodeURIComponent(title)}
                statusText={null}
                statusContent={this.renderTooltip()}
                aria-label={`${title} ${msgs.get('secondaryHeader', locale)}`}
              >
                {breadcrumbItems && (
                  <Breadcrumb>{this.renderBreadCrumb()}</Breadcrumb>
                )}
              </DetailPageHeader>
              {this.state.shadowPresent && (
                <React.Fragment>
                  <div
                    className={
                      breadcrumbItems
                        ? 'header-box-shadow--tall'
                        : 'header-box-shadow'
                    }
                  />
                </React.Fragment>
              )}
              {tabs &&
                tabs.length > 0 && (
                  <Tabs
                    className={
                      breadcrumbItems ? 'cluster-tabs--long' : 'cluster-tabs'
                    }
                    selected={this.getSelectedTab() || 0}
                    aria-label={`${title} ${msgs.get('tabs.label', locale)}`}
                  >
                    {this.renderTabs()}
                  </Tabs>
              )}
            </React.Fragment>
            {actions && this.renderActions()}
          </div>
          {links &&
            links.length > 0 && (
              <div className="secondary-header-links">{this.renderLinks()}</div>
          )}
        </div>
      )
    } else {
      return (
        <div
          className="secondary-header-wrapper-min"
          role="region"
          aria-label={`${title} ${msgs.get('secondaryHeader', locale)}`}
        >
          <div className="secondary-header simple-header">
            <h1 className="bx--detail-page-header-title">
              {decodeURIComponent(title)}
            </h1>
            {this.renderTooltip()}
          </div>
        </div>
      )
    }
  }

  renderBreadCrumb() {
    const { breadcrumbItems } = this.props
    return (
      breadcrumbItems &&
      breadcrumbItems.map((breadcrumb, index) => {
        const key = `${breadcrumb}-${index}`
        return (
          <React.Fragment key={key}>
            <div
              className="bx--breadcrumb-item"
              title={decodeURIComponent(breadcrumb.label)}
            >
              <Link to={breadcrumb.url} className="bx--link">
                {decodeURIComponent(breadcrumb.label)}
              </Link>
            </div>
          </React.Fragment>
        )
      })
    )
  }

  renderLinks() {
    const { links } = this.props,
          { locale } = this.context
    return links.map(link => {
      const {
        id,
        label,
        url,
        kind = 'primary',
        title,
        handleClick = () => this.props.history.push(url)
      } = link
      // if portal, react component will create the button using a portal
      if (kind === 'portal') {
        return !title ? <div key={id} id={id} className="portal" /> : null
      }
      return (
        <Button key={id} id={id} onClick={handleClick} kind={kind}>
          {msgs.get(label, locale)}
        </Button>
      )
    })
  }

  renderActions() {
    const { actions } = this.props
    return (
      <div className="secondary-header-actions">
        <Button
          kind="secondary"
          onClick={() => actions.secondary && actions.secondary.action()}
          className="secondary-header-actions-secondary"
        >
          {actions.secondary && actions.secondary.label}
        </Button>
        <Button
          kind="primary"
          onClick={() => actions.primary && actions.primary.action()}
          disabled={actions.primary.disabled}
          className="secondary-header-actions-primary"
        >
          {actions.primary && actions.primary.label}
        </Button>
      </div>
    )
  }

  renderTabs() {
    const { tabs } = this.props,
          { locale } = this.context
    return tabs.map(tab => {
      return (
        <Tab
          label={msgs.get(tab.label, locale)}
          key={tab.id}
          id={tab.id}
          href={tab.url}
          onClick={
            tab.handleClick
              ? tab.handleClick
              : this.clickTab.bind(this, tab.url)
          }
        />
      )
    })
  }

  renderTooltip() {
    const { tooltip } = this.props
    const { locale } = this.context
    const { links = [] } = this.props
    return (
      <React.Fragment>
        {tooltip && (
          <SecondaryHeaderTooltip
            text={tooltip.text}
            link={tooltip.link}
            linkText={msgs.get('tooltip.link', locale)}
          />
        )}
        {links &&
          links.map(link => {
            const { id, kind, title } = link
            // if portal, react component will create the button using a portal
            if (kind === 'portal' && title) {
              return <div key={id} id={id} className="portal" />
            } else {
              return null
            }
          })}
      </React.Fragment>
    )
  }

  getSelectedTab() {
    const { tabs, location } = this.props
    const selectedTab = tabs
      .map((tab, index) => {
        tab.index = index
        return tab
      })
      .filter((tab, index) => {
        if (index === 0) {
          return false
        }
        return location.pathname.startsWith(tab.url)
      })
    return selectedTab[0] && selectedTab[0].index
  }

  clickTab(url) {
    this.props.history.replace(url)
  }
}

SecondaryHeader.propTypes = {
  actions: PropTypes.array,
  breadcrumbItems: PropTypes.array,
  history: PropTypes.object,
  links: PropTypes.array,
  location: PropTypes.object,
  tabs: PropTypes.array,
  title: PropTypes.string,
  tooltip: PropTypes.string
}

SecondaryHeader.contextTypes = {
  locale: PropTypes.string
}

const mapStateToProps = state => {
  return {
    title: state.secondaryHeader.title,
    tabs: state.secondaryHeader.tabs,
    actions: state.secondaryHeader.actions,
    breadcrumbItems: state.secondaryHeader.breadcrumbItems,
    links: state.secondaryHeader.links,
    tooltip: state.secondaryHeader.tooltip,
    role: state.role && state.role.role
  }
}

export default withRouter(connect(mapStateToProps)(SecondaryHeader))
