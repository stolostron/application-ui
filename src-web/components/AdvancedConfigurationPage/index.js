/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.

import React from 'react'
import msgs from '../../../nls/platform.properties'
import { connect } from 'react-redux'
import resources from '../../../lib/shared/resources'
import {
  updateSecondaryHeader,
  clearSuccessFinished
} from '../../actions/common'

import ApplicationDeploymentHighlights from '../ApplicationDeploymentHighlights'
import { withRouter } from 'react-router-dom'
import AdvancedConfigurationLists from '../AdvancedConfigurationLists'

/* eslint-disable react/prop-types */

resources(() => {
  require('./style.scss')
})

const mapDispatchToProps = dispatch => {
  return {
    clearSuccessFinished: () => clearSuccessFinished(dispatch),
    updateSecondaryHeaderFn: (title, tabs, mainButton) =>
      dispatch(
        updateSecondaryHeader(title, tabs, null, null, null, null, mainButton)
      )
  }
}

class AdvancedConfigurationPage extends React.Component {
  componentDidMount() {
    const {
      secondaryHeaderProps,
      updateSecondaryHeaderFn,
      locale
    } = this.props

    updateSecondaryHeaderFn(
      msgs.get(secondaryHeaderProps.title, locale),
      secondaryHeaderProps.tabs,
      secondaryHeaderProps.mainButton
    )
  }

  componentWillUnmount() {
    this.mutateFinished()
  }

  mutateFinished() {
    this.props.clearSuccessFinished()
  }

  render() {
    const { secondaryHeaderProps } = this.props

    return (
      <React.Fragment>
        <ApplicationDeploymentHighlights />
        <AdvancedConfigurationLists
          secondaryHeaderProps={secondaryHeaderProps}
        />
      </React.Fragment>
    )
  }
}

export default withRouter(
  connect(null, mapDispatchToProps)(AdvancedConfigurationPage)
)
