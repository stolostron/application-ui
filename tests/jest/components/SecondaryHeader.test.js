/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import { SecondaryHeader } from "../../../src-web/components/SecondaryHeader";

describe("SecondaryHeader component 1", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it("renders as expected with window size < 1200", () => {
    window.innerWidth = 1000;
    const navToggle = {
      addEventListener: jest.fn
    };
    const navBar = {
      className: "pf-m-expanded"
    };
    const secHeader = {
      style: {
        width: 1200
      }
    };

    jest.spyOn(document, "querySelector").mockImplementation(selector => {
      switch (selector) {
        case "#nav-toggle":
          return navToggle;
        case ".pf-c-page__sidebar":
          return navBar;
        case ".secondary-header":
          return secHeader;
      }
    });
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(document.querySelector).toBeCalledTimes(3);
  });

  it("renders as expected with window size > 1200", () => {
    window.innerWidth = 1300;
    const navToggle = {
      addEventListener: jest.fn
    };
    const navBar = {
      className: "pf-m-expanded"
    };
    const secHeader = {
      style: {
        width: 1200
      }
    };

    jest.spyOn(document, "querySelector").mockImplementation(selector => {
      switch (selector) {
        case "#nav-toggle":
          return navToggle;
        case ".pf-c-page__sidebar":
          return navBar;
        case ".secondary-header":
          return secHeader;
      }
    });
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(document.querySelector).toBeCalledTimes(3);
  });

  it("renders as expected with window size > 1200, navBar is not expanded", () => {
    window.innerWidth = 1300;
    const navToggle = {
      addEventListener: jest.fn
    };
    const navBarNotExpanded = {
      className: "pf-m-not-expanded"
    };
    const secHeader = {
      style: {
        width: 1200
      }
    };

    jest.spyOn(document, "querySelector").mockImplementation(selector => {
      switch (selector) {
        case "#nav-toggle":
          return navToggle;
        case ".pf-c-page__sidebar":
          return navBarNotExpanded;
        case ".secondary-header":
          return secHeader;
      }
    });
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(document.querySelector).toBeCalledTimes(3);
  });
});

describe("SecondaryHeader component 2", () => {
  const tabs = [
    {
      id: "dashboard-application",
      label: "tabs.dashboard.application",
      url: "/multicloud/dashboard"
    }
  ];
  it("renders as expected", () => {
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" tabs={tabs} />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("SecondaryHeader component 3", () => {
  const tabs = [
    {
      id: "logs-tab1",
      label: "tabs.dashboard.application",
      url: "/multicloud/dashboard"
    },
    {
      id: "logs-tab2",
      label: "tabs.dashboard",
      url: "/hello"
    }
  ];
  const location = {
    pathname: "/hello"
  };
  it("renders as expected", () => {
    const component = renderer.create(
      //eslint-disable-next-line
      <BrowserRouter>
        <SecondaryHeader title="hello world" tabs={tabs} location={location} />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
