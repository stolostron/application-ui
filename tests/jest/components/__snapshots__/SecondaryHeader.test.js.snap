// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeader component 1 renders as expected 1`] = `
<div
  id="ApplicationSecondaryHeader"
>
  <div
    aria-label="hello world secondary"
    className="secondary-header-wrapper-min"
    role="region"
  >
    <div
      className="secondary-header simple-header"
    >
      <h1
        className="bx--detail-page-header-title"
      >
        hello world
      </h1>
    </div>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders as expected 1`] = `
<div
  id="ApplicationSecondaryHeader"
>
  <div
    aria-label="hello world"
    className="secondary-header-wrapper"
    role="region"
  >
    <div
      className="secondary-header simple-header"
    >
      <header
        aria-label="Heading: hello world"
      >
        <div
          className="bx--detail-page-header-content"
        >
          <header
            aria-label="hello world secondary"
            className="bx--detail-page-header bx--detail-page-header--with-tabs"
            data-header-active={false}
            role="banner"
          >
            <div
              className="bx--detail-page-header-content"
            >
              <div
                className="bx--breadcrumb"
              />
              <div
                className="bx--detail-page-header-title-container"
              >
                <div
                  className="bx--detail-page-header-icon-container"
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="#D8D8D8"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
                <h1
                  className="bx--detail-page-header-title"
                >
                  hello world
                </h1>
                <div
                  className="bx--detail-page-header-status-container"
                >
                  <div
                    className="bx--detail-page-header-status-icon"
                    style={
                      Object {
                        "backgroundColor": undefined,
                      }
                    }
                  />
                   
                  <span
                    className="bx--detail-page-header-status-text"
                  >
                    Running
                  </span>
                </div>
              </div>
              <nav
                aria-label="hello world undefined"
                className="bx--tabs"
                role="navigation"
                selected={0}
              >
                <div
                  aria-label="listbox"
                  className="bx--tabs-trigger"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <a
                    className="bx--tabs-trigger-text"
                    href="#"
                    onClick={[Function]}
                    tabIndex={-1}
                  >
                    Application
                  </a>
                  <svg
                    alt="show menu options"
                    aria-label="show menu options"
                    fillRule="evenodd"
                    height="5"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      show menu options
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </div>
                <ul
                  className="bx--tabs__nav bx--tabs__nav--hidden"
                  role="tablist"
                >
                  <li
                    className="bx--tabs__nav-item bx--tabs__nav-item--selected"
                    id="dashboard-application"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="presentation"
                    selected={true}
                    tabIndex={-1}
                  >
                    <a
                      aria-selected={true}
                      className="bx--tabs__nav-link"
                      href="/multicloud/dashboard"
                      role="tab"
                      tabIndex={0}
                    >
                      Application
                    </a>
                  </li>
                </ul>
              </nav>
              <div
                aria-hidden={false}
                className="tab-content"
                hidden={false}
                selected={true}
              />
            </div>
            <div
              className="bx--detail-page-header-menu"
            />
          </header>
        </div>
      </header>
    </div>
  </div>
</div>
`;

exports[`SecondaryHeader component 3 renders as expected 1`] = `
<div
  id="ApplicationSecondaryHeader"
>
  <div
    aria-label="hello world"
    className="secondary-header-wrapper"
    role="region"
  >
    <div
      className="secondary-header simple-header"
    >
      <header
        aria-label="Heading: hello world"
      >
        <div
          className="bx--detail-page-header-content"
        >
          <header
            aria-label="hello world secondary"
            className="bx--detail-page-header bx--detail-page-header--with-tabs"
            data-header-active={false}
            role="banner"
          >
            <div
              className="bx--detail-page-header-content"
            >
              <div
                className="bx--breadcrumb"
              />
              <div
                className="bx--detail-page-header-title-container"
              >
                <div
                  className="bx--detail-page-header-icon-container"
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="#D8D8D8"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
                <h1
                  className="bx--detail-page-header-title"
                >
                  hello world
                </h1>
                <div
                  className="bx--detail-page-header-status-container"
                >
                  <div
                    className="bx--detail-page-header-status-icon"
                    style={
                      Object {
                        "backgroundColor": undefined,
                      }
                    }
                  />
                   
                  <span
                    className="bx--detail-page-header-status-text"
                  >
                    Running
                  </span>
                </div>
              </div>
              <nav
                aria-label="hello world undefined"
                className="bx--tabs"
                role="navigation"
                selected={1}
              >
                <div
                  aria-label="listbox"
                  className="bx--tabs-trigger"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <a
                    className="bx--tabs-trigger-text"
                    href="#"
                    onClick={[Function]}
                    tabIndex={-1}
                  >
                    provide a label
                  </a>
                  <svg
                    alt="show menu options"
                    aria-label="show menu options"
                    fillRule="evenodd"
                    height="5"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      show menu options
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </div>
                <ul
                  className="bx--tabs__nav bx--tabs__nav--hidden"
                  role="tablist"
                >
                  <li
                    className="bx--tabs__nav-item"
                    id="logs-tab1"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="presentation"
                    selected={false}
                    tabIndex={-1}
                  >
                    <a
                      aria-selected={false}
                      className="bx--tabs__nav-link"
                      href="/multicloud/dashboard"
                      role="tab"
                      tabIndex={0}
                    >
                      Application
                    </a>
                  </li>
                  <li
                    className="bx--tabs__nav-item bx--tabs__nav-item--selected"
                    id="logs-tab2"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="presentation"
                    selected={true}
                    tabIndex={-1}
                  >
                    <a
                      aria-selected={true}
                      className="bx--tabs__nav-link"
                      href="/hello"
                      role="tab"
                      tabIndex={0}
                    >
                      provide a label
                    </a>
                  </li>
                </ul>
              </nav>
              <div
                aria-hidden={true}
                className="tab-content"
                hidden={true}
                selected={false}
              />
              <div
                aria-hidden={false}
                className="tab-content"
                hidden={false}
                selected={true}
              />
            </div>
            <div
              className="bx--detail-page-header-menu"
            />
          </header>
        </div>
      </header>
    </div>
  </div>
</div>
`;
